packages/taquito/test/prepare/prepare-provider.spec.ts

```diff
@@ -0,0 +1,1000 @@
+      expect(res).toEqual(preparedOriginationOpWithReveal);
+    });
+
+    it('should be able to prepare origination op without reveal op when reveal estimate returns undefined', async () => {
+      jest.spyOn(context.estimate, 'reveal').mockResolvedValue(undefined);
+
+      const prepared = await prepareProvider.originate({
+        balance: '1',
+        code: `parameter string;
+        storage string;
+        code {CAR;
+              PUSH string "Hello ";
+              CONCAT;
+              NIL operation; PAIR};
+        `,
+        init: `"test"`,
+      });
+
+      const res = JSON.parse(JSON.stringify(prepared));
+
+      expect(res).toEqual(preparedOriginationOpNoReveal);
+    });
+  });
+
+  describe('transaction', () => {
+    it('should return a prepared Transaction operation with a reveal op', async () => {
+      jest.spyOn(context.estimate, 'reveal').mockResolvedValue(estimate);
+
+      const prepared = await prepareProvider.transaction({
+        to: 'tz1QZ6KY7d3BuZDT1d19dUxoQrtFPN2QJ3hn',
+        amount: 2,
+      });
+
+      expect(prepared).toEqual({
+        opOb: {
+          branch: 'test_block_hash',
+          contents: [
+            {
+              kind: 'reveal',
+              fee: '391',
+              public_key: 'test_pub_key',
+              source: 'test_public_key_hash',
+              gas_limit: '101',
+              storage_limit: '1000',
+              counter: '1',
+            },
+            {
+              kind: 'transaction',
+              fee: '391',
+              gas_limit: '101',
+              storage_limit: '1000',
+              amount: '2000000',
+              destination: 'tz1QZ6KY7d3BuZDT1d19dUxoQrtFPN2QJ3hn',
+              source: 'test_public_key_hash',
+              counter: '2',
+            },
+          ],
+          protocol: 'test_protocol',
+        },
+        counter: 0,
+      });
+    });
+
+    it('should be able to prepare transaction op without reveal op when estimate returns undefined', async () => {
+      jest.spyOn(context.estimate, 'reveal').mockResolvedValue(undefined);
+
+      const prepared = await prepareProvider.transaction({
+        to: 'tz1QZ6KY7d3BuZDT1d19dUxoQrtFPN2QJ3hn',
+        amount: 2,
+      });
+
+      expect(prepared).toEqual({
+        opOb: {
+          branch: 'test_block_hash',
+          contents: [
+            {
+              kind: 'transaction',
+              fee: '391',
+              gas_limit: '101',
+              storage_limit: '1000',
+              amount: '2000000',
+              destination: 'tz1QZ6KY7d3BuZDT1d19dUxoQrtFPN2QJ3hn',
+              source: 'test_public_key_hash',
+              counter: '1',
+            },
+          ],
+          protocol: 'test_protocol',
+        },
+        counter: 0,
+      });
+    });
+  });
+
+  describe('drainDelegate', () => {
+    it('should return a prepared drain_delegate operation', async () => {
+      jest.spyOn(context.estimate, 'reveal').mockResolvedValue(estimate);
```

Comment:
I am unsure if my understanding is correct, would like to double-check. Is estimation step only necessary for operations with `gas_limit`, `storage_limit`, `fee` properties in contents object? This is the reason when I worked with drain_delegate, I didn't update it in `EstimationProvider`. If my assumption is correct, maybe we don't need line 251, but I can be wrong and didn't implement drain_delegate correctly too.

Reply:
or line 251 supposed to be `.mockResolveValue(undefined)`?

Reply:
Yeah you are correct, it was a miss on my part. Thanks for catching that